packages/core/src/hooks/useSupportConfig.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useApi, configApiRef } from '@backstage/core-api';
+
+export type SupportItemLink = {
+  url: string;
+  title: string;
+};
+
+export type SupportItem = {
+  title: string;
+  icon?: string;
+  links?: SupportItemLink[];
+};
+
+export type SupportConfig = {
+  url: string;
+  items: SupportItem[];
+};
+
+export function useSupportConfig(): SupportConfig {
+  const config = useApi(configApiRef);
+  const supportConfig = config.getOptional('app.support') as SupportConfig;
+
```

Comment:
I just realized that this doesn't do any validation on the input; it just trusts that the user has gotten the entire structure right. If they don't, some code will break far away elsewhere for unclear reasons.

I think this is in an excellent state otherwise, so let's merge and tweak later so I don't block you even more with slow reviewing. Thanks for your patience. If you get the time later, you can use the full suite of getOptionalConfig and then a bunch of getString and stuff to extract all config and build the output object from it. That way we get the error checking in place.

Yes this could be made more convenient if we built some handy schema validation feature into it or something.

Reply:
That's a good point. I actually had this problem earlier during development. I've updated the code to build out the structure based using the appropriate config methods. I also make `links` required as it doesn't make much sense for the array to be empty.